taskmanager/src/Presentation/Controller/TaskController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Presentation\Controller;
+
+use App\Application\Command\CreateTaskCommand;
+use App\Application\Query\AllTasksQuery;
+use App\Application\ServiceBus\CommandBus;
+use App\Application\ServiceBus\QueryBus;
+use App\Infrastructure\Adapters\Incoming\JSON\AddTask;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+class TaskController extends Controller
+{
+    /**
+     * @Route("/")
+     */
+    public function listAction()
+    {
+        $query = new AllTasksQuery();
+        /** @var QueryBus $queryBus */
+        $queryBus = $this->container->get('application.query_bus');
+        $tasks = $queryBus->handle($query);
+
+        return JsonResponse::create($tasks);
+    }
+
+    /**
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     * @Route("/new")
+     */
+    public function createAction(Request $request)
+    {
+        $data = $request->getContent();
+
+        // TODO Need validation data
+
+        $command = new CreateTaskCommand($data);
```

Comment:
IMHO using commands as a DTO without any control can be risky and may occur a lot of unexpected issues. My recommendation is:
1. Use Symfony Form
2. Make properties public but use `Symfony Form & Validator` to validate data
3. No more unspecified properties

and questions that come here:
1. How to return ID of Task?
2. How to validate date? 
3. How to handle logs?
4. What would happen if I switch to async command bus like RabbitMQ?

Reply:
Yes, there a lots of things to do, `Symfony Form & Validator` is prefer solution for this. We don't lost control - "unexpected issues" should throw exception so we can take care about it. About questions:
ad 1. I don't want broke CQRS convention so commands nothing comes back. There is couple way to do this - for example we can create UUID on client side and send in POST request.  
ad 2. What do you mean about "validate date" ?
ad 3. Good place for handle logs can be Query/CommandBus
ad 4. To be honest I don't think about this yet

Btw. CQRS is additional things  - we can use adapters directly in controllers.